MVC-App/Services/RelationService.cs

```diff
@@ -0,0 +1,280 @@
+                        {
+                            if (char.IsDigit(value[valueIterator]))
+                            {
+                                result.Add(value[valueIterator]);
+
+                                valueIterator++;
+                            }
+                            else return value;
+
+                            break;
+                        }
+                    case 'L':
+                        {
+                            if (char.IsLetter(value[valueIterator]))
+                            {
+                                result.Add(char.ToUpper(value[valueIterator]));
+
+                                valueIterator++;
+                            }
+                            else return value;
+
+                            break;
+                        }
+                    case 'l':
+                        {
+                            if (char.IsLetter(value[valueIterator]))
+                            {
+                                result.Add(char.ToLower(value[valueIterator]));
+
+                                valueIterator++;
+                            }
+                            else return value;
+
+                            break;
+                        }
+                    default:
+                        {
+                            result.Add(mask[i]);
+                            if (!char.IsLetterOrDigit(value[valueIterator]))
+                            {
+                                if (value[valueIterator] == mask[i])
+                                {
+                                    valueIterator++;
+                                }
+                                else return value;
+                            }
+                            break;
+                        }
+                }
+            }
+            if (value.Length == valueIterator)
+            {
+                return new string(result.ToArray());
+            }
+            else return value;
+        }
+
+        public async Task<Relation> GetRelationAsync(Guid? id)
+        {
+            return _unitOfWork.RelationRepository.GetByID(id);
+        }
+
+        public async Task<RelationListVM> GetAsync(Guid? categoryId)
+        {
+            var countries = _unitOfWork.CountryRepository.Get().ToList();
+
+            var relationModels = await InitRelationModels();
+
+            if (categoryId != null && categoryId != Guid.Empty)
+            {
+                relationModels = relationModels.Where(p => p.Categories.Contains(categoryId.Value)).ToList();
+            }
+
+            var relationListVM = new RelationListVM { RelationViewModels = relationModels, Categories = Categories, Countries = countries };
+
+            return relationListVM;
+        }
+
+        public async Task Create(CreateEditRelationVM relationVM)
+        {
+            var relation = new Relation
+            {
+                Id = Guid.NewGuid(),
```

Comment:
You can check if can you use AutoMapper here to map models from CreateEditRelationVM to Relation
You can implement in in separate task